src/rgw/driver/posix/rgw_sal_posix.cc

```diff
@@ -58,6 +58,91 @@ static inline void object_statx_save(struct statx& stx, RGWObjState& state)
   state.mtime = ceph::real_clock::from_time_t(stx.stx_mtime.tv_sec);
 }
 
+static int get_x_attrs(optional_yield y, const DoutPrefixProvider* dpp, int fd,
+		       Attrs& attrs, const std::string& display)
+{
+  char namebuf[64 * 1024]; // Max list size supported on linux
+  ssize_t buflen;
+  int ret;
+
+  buflen = flistxattr(fd, namebuf, sizeof(namebuf));
+  if (buflen < 0) {
+    ret = errno;
+    ldpp_dout(dpp, 0) << "ERROR: could not list attributes for " << display << ": "
+      << cpp_strerror(ret) << dendl;
+    return -ret;
+  }
+
+  char *keyptr = namebuf;
+  while (buflen > 0) {
+    std::string value;
+    ssize_t vallen, keylen;
+    char* vp;
+
+    keylen = strlen(keyptr) + 1;
+    std::string key(keyptr);
```

Comment:
could you use a string_view here?  It even has builtins for slicing prefix, suffix substrings.

Reply:
I'll look into it.

Reply:
I'm emplacing key below (at line 125) so I think that means I can't use a string_view